### ISSUE TITLE
Many different DSC of FLARE23 and TotalSegmentor

### ISSUE THREAD

**Issue by Itsanewday** [MrGiovanni/SuPreM#10]
Thanks for sharing the great AbdomenAtlas1.1 and many pretrained models.
Here are my questions about the paper.
First, I think that the performance comparison(0.895) on FLARE23 maybe not proper. Here is the [final test results of FLARE23](https://codalab.lisn.upsaclay.fr/competitions/12239#learn_the_details-testing_results), at least 4 teams get 0.91 or higher average Organs DSC, and the best is about 0.93. Besides, according to the [summary of FLARE22](https://link.springer.com/book/10.1007/978-3-031-23911-3 ) most of them have very good robustness. Did you finetune on FLARE23 and how about the performance? 
The second question is about Fine-tuning on novel classes.  As mentioned in the paper, there are muscles, cardiac structures, organs, vertebrae and tumors. I wonder that how many models did you use? Did you re-train a model from scratch? In my opinion, the specific model maybe better than the universal model, even the later is pretrained on much larger dataset. 
The last is about the scratch performance on TotalSegmentor. In tabel 7 the scratch DSC is different from the results of Totalsegmentor paper. In fact, the totalsegmentor has about 1.5 higher average dsc than SuPreM. [Details results of totalsegmentorV2](https://github.com/wasserth/TotalSegmentator/blob/master/resources/evaluate_results.txt) and [V1](https://github.com/wasserth/TotalSegmentator/blob/master/resources/results_all_classes_v1.json)

**Comment by WenxuanChelsea** [comment]
Hi @Itsanewday 

Thanks for your interest in our work!

Regarding your questions:
> Q1. The performance comparison(0.895) on FLARE23 maybe not proper. Did you finetune on FLARE23 and how about the performance?

At the time the paper was written, the test results were not released by FLARE-2023, so we simply reported the performance that we obtained (0.895)—not many training tricks were used. In the paper, we only reported the direct inference results on FLARE-2023 to check the generalizability of SuPreM, no fine-tuning is involved.

>Q2. I wonder that how many models did you use? Did you re-train a model from scratch?

We consider organs, muscles, vertebrae, cardiac structures, and tumors as different tasks, so five specific models were trained—one for each task. We also report training-from-scratch performance for each task in Table 2 (https://www.cs.jhu.edu/~alanlab/Pubs23/li2023suprem.pdf), suggesting that supervised pre-training on organ masks greatly benefits segmenting novel classes (other structures/tumors) in each task.

>Q3. The scratch performance on TotalSegmentor.

This question was also raised by the reviewer at ICLR (https://openreview.net/forum?id=AhizIPytk4). We bring our response here:

In TotalSegmentator, the labels were largely generated by a **single** nnU-Net re-trained continually (see Figure 1b in [Wasserthal et al.](https://pubs.rsna.org/doi/full/10.1148/ryai.230024)). Depending solely on nnU-Net could introduce a potential label bias favoring the nnU-Net architecture. This means two points. 

1. Their ground truth (revised pseudo labels) could be biased to the nnU-Net architecture. nnU-Net trained and tested on this dataset will achieve an **unreachable** performance as shown in Totalsegmentator [github](https://github.com/wasserth/TotalSegmentator/blob/master/resources/evaluate_results.txt). For example, they report the DSC score of 0.894 for pancreas segmentation, this number has never been reached literature. The MSD top 1 result (**[0.828](https://decathlon-10.grand-challenge.org/evaluation/challenge/leaderboard/)**); TCIA Pancreas-CT Dataset top 1 result (**[0.845](https://paperswithcode.com/sota/pancreas-segmentation-on-tcia-pancreas-ct)**); even the FELIX Project (producing the largest pancreas dataset in USA) only achieves **[0.87](https://www.medrxiv.org/content/10.1101/2022.09.24.22280071v1)** DSC score. None of these advanced benchmarks have achieved 0.894 reported in Totalsegmentator. Therefore, it is sensible to assume the ground truth is biased to the nnU-Net architecture (including iliopsoas that you mentioned and many other classes). This is the reason why we do not make comparisons with Totalsegmentator. 

2. Due to the potential label biases, whenever TotalSegmentator is employed for benchmarking, nnU-Net and models building upon nnU-Net would always outperform other segmentation architectures (e.g., UNETR, TransUNet, SwinUNet, etc.). This observation has also been made in several publications that used the TotalSegmentator dataset. For example, Table 3 in [[Huang et al., 2023](https://arxiv.org/abs/2304.06716)] showed that nnFormer, UNETR, and Swin UNETR were all outperformed by nnU-Net and models building upon nnU-Net in TotalSegmentator. More importantly, the average DSC achieved by our model on TotalSegmentator is also much higher than [[Huang et al., 2023](https://arxiv.org/abs/2304.06716)] as compared in the following table.

| method &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;  | organ &nbsp; &nbsp; &nbsp; | vertebrae &nbsp; &nbsp; &nbsp; | cardiac &nbsp; &nbsp; &nbsp; | muscle &nbsp; &nbsp; &nbsp; | 
|  ----  | ----  |  ----  |  ----  |  ----  | 
| [Huang et al., 2023](https://arxiv.org/abs/2304.06716) | 89.82 | 90.43 | 90.89 | 88.83 |
| Ours | **92.09** | **91.29** | **92.21** | **95.40** | 

Therefore, we believe that the segmentation results in TotalSegmentator reported in our paper are **compelling** and should be considered a **faithful** benchmark.

Thanks,
Wenxuan

**Comment by Itsanewday** [comment]
Thanks for the detailed reply！



---Original---
From: "Wenxuan ***@***.***&gt;
Date: Sun, May 12, 2024 11:59 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [MrGiovanni/SuPreM] Many different DSC of FLARE23 andTotalSegmentor (Issue #10)




 
Hi @Itsanewday
 
Thanks for your interest in our work!
 
Regarding your questions:
  
Q1. The performance comparison(0.895) on FLARE23 maybe not proper. Did you finetune on FLARE23 and how about the performance?
  
At the time the paper was written, the test results were not released by FLARE-2023, so we simply reported the performance that we obtained (0.895)—not many training tricks were used. In the paper, we only reported the direct inference results on FLARE-2023 to check the generalizability of SuPreM, no fine-tuning is involved.
  
Q2. I wonder that how many models did you use? Did you re-train a model from scratch?
  
We consider organs, muscles, vertebrae, cardiac structures, and tumors as different tasks, so five specific models were trained—one for each task. We also report training-from-scratch performance for each task in Table 2 (https://www.cs.jhu.edu/~alanlab/Pubs23/li2023suprem.pdf), suggesting that supervised pre-training on organ masks greatly benefits segmenting novel classes (other structures/tumors) in each task.
  
Q3. The scratch performance on TotalSegmentor.
  
This question was also raised by the reviewer at ICLR (https://openreview.net/forum?id=AhizIPytk4). We bring our response here:
 
In TotalSegmentator, the labels were largely generated by a single nnU-Net re-trained continually (see Figure 1b in Wasserthal et al.). Depending solely on nnU-Net could introduce a potential label bias favoring the nnU-Net architecture. This means two points.
  
 
Their ground truth (revised pseudo labels) could be biased to the nnU-Net architecture. nnU-Net trained and tested on this dataset will achieve an unreachable performance as shown in Totalsegmentator github. For example, they report the DSC score of 0.894 for pancreas segmentation, this number has never been reached literature. The MSD top 1 result (0.828); TCIA Pancreas-CT Dataset top 1 result (0.845); even the FELIX Project (producing the largest pancreas dataset in USA) only achieves 0.87 DSC score. None of these advanced benchmarks have achieved 0.894 reported in Totalsegmentator. Therefore, it is sensible to assume the ground truth is biased to the nnU-Net architecture (including iliopsoas that you mentioned and many other classes). This is the reason why we do not make comparisons with Totalsegmentator.
 
 
 
Due to the potential label biases, whenever TotalSegmentator is employed for benchmarking, nnU-Net and models building upon nnU-Net would always outperform other segmentation architectures (e.g., UNETR, TransUNet, SwinUNet, etc.). This observation has also been made in several publications that used the TotalSegmentator dataset. For example, Table 3 in [Huang et al., 2023] showed that nnFormer, UNETR, and Swin UNETR were all outperformed by nnU-Net and models building upon nnU-Net in TotalSegmentator. More importantly, the average DSC achieved by our model on TotalSegmentator is also much higher than [Huang et al., 2023] as compared in the following table.
 
     method &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; organ &nbsp; &nbsp; &nbsp; vertebrae &nbsp; &nbsp; &nbsp; cardiac &nbsp; &nbsp; &nbsp; muscle &nbsp; &nbsp; &nbsp; 
    Huang et al., 2023 89.82 90.43 90.89 88.83 
  Ours 92.09 91.29 92.21 95.40 
   
Therefore, we believe that the segmentation results in TotalSegmentator reported in our paper are compelling and should be considered a faithful benchmark.
 
Thanks,
 Wenxuan
